src/lib/alloc.c

```diff
@@ -470,7 +518,9 @@ void *rmalloc(int zone, uint32_t caps, size_t bytes)
 		trace_mem_error("rmalloc() error: invalid zone");
 		break;
 	}
-
+#if DEBUG_BLOCK_FREE
+	bzero(ptr, bytes);
+#endif
```

Comment:
Memory is filled with with the pattern at the fw startup if we don't zero it out when we allocate it we will provide trash for any allocated memory and it will crash FW. We would have to handle it in code in every allocation. That is why I would strongly suggest to change allocation so that every allocated memory is passed as zeroed. @mmaka1 @lgirdwood What do You think?

Reply:
Oh, I got your idea now. 
We need check ptr is NULL or not here.
And what about rballoc()?